src/session/SessionManager.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright (C) 2006 - 2017 Evan Teran
+                          evan.teran@gmail.com
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "SessionManager.h"
+#include "IPlugin.h"
+#include "QJsonDocument.h"
+#include "QJsonObject.h"
+#include "QJsonParseError.h"
+#include "edb.h"
+
+#include <QDateTime>
+#include <QDebug>
+#include <QFile>
+#include <QMessageBox>
+
+const int  SessionFileVersion  = 1;
+const auto SessionFileIdString = QLatin1String("edb-session");
+
+SessionManager& SessionManager::instance() {
+  static SessionManager inst;
+  return inst;
+}
+
+//------------------------------------------------------------------------------
+// Name: load_session
+// Desc:
+//------------------------------------------------------------------------------
+bool SessionManager::load_session(const QString &session_file, SessionError& session_error) {
+
+  QFile file(session_file);
+	if(file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+		QByteArray json = file.readAll();
+		QJsonParseError error;
+		auto doc = QJsonDocument::fromJson(json, &error);
+		if(error.error != QJsonParseError::NoError) {
+      session_error.err = SessionError::UnknownError;
```

Comment:
Indentation appears broken here.

Reply:
Weird, this seems to be a problem with my IDE (everything is fine there) but VIM is showing the same indentation error... Sorry

Reply:
It's due to tab size. On github it's 8 chars by default.

Reply:
This just means one should be consistent in using tabs vs spaces for indentation (in EDB the syle is to prefer tabs).

Reply:
Looked into the settings of my IDE and the tab size is 2 spaces. The indentation comes from my IDE trying to figure out the tab size of the moved code. Should i change the size in my IDE (found nothing in the style guide)?

Reply:
Ideally, moved code should be moved verbatim, including any whitespace. It might be impossible to accomplish depending on the editor (although Vim should be able to do it, I've never had any problem with its yank-put functions).
As for setting up the IDE, I suppose you want to set tab policy to "never expand" and tab size for display to 4 chars (for Vim it's `set noexpandtab` and `set ts=4` respectively).